executor/oci/hosts.go

```diff
@@ -47,7 +48,11 @@ func makeHostsFile(stateDir string, extraHosts []executor.HostIP, idmap *idtools
 
 	b := &bytes.Buffer{}
 
-	if _, err := b.Write([]byte(hostsContent)); err != nil {
+	var hosts = hostsContent
+	if hostname != "" {
+		hosts = strings.Replace(hosts, "buildkitsandbox", hostname, 1)
```

Comment:
`strings.Replace()` doesn't look robust.

`const hostsContent` should be changed to a function that takes hostname as a argument.

Reply:
Wrapped to `initHostsFile()`, please help to review~